eos/db/saveddata/databaseRepair.py

```diff
@@ -160,3 +160,75 @@ def NullTargetResistNames(saveddata_engine):
             query = "UPDATE 'targetResists' SET 'name' = 'Unknown' WHERE name IS NULL OR name = ''"
             delete = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
             logger.error("Database corruption found. Cleaning up %d records.", delete.rowcount)
+
+    @staticmethod
+    def OrphanedFitIDItemID(saveddata_engine):
+        # Orphaned items that are missing the fit ID or item ID.
+        # See issue #954
+        for table in ['drones', 'cargo', 'fighters']:
+            logger.debug("Running database cleanup for orphaned %s items.", table)
+            query = "SELECT COUNT(*) AS num FROM " + table + " WHERE itemID IS NULL OR itemID = '' or itemID = '0' or fitID IS NULL OR fitID = '' or fitID = '0'"
+            results = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
+
+            if results is None:
+                return
+
+            row = results.first()
+
+            if row and row['num']:
+                query = "DELETE FROM " + table + " WHERE itemID IS NULL OR itemID = '' or itemID = '0' or fitID IS NULL OR fitID = '' or fitID = '0'"
+                delete = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
+                logger.error("Database corruption found. Cleaning up %d records.", delete.rowcount)
+
+        for table in ['modules']:
+            logger.debug("Running database cleanup for orphaned %s items.", table)
+            query = "SELECT COUNT(*) AS num FROM " + table + " WHERE itemID = '0' or fitID IS NULL OR fitID = '' or fitID = '0'"
+            results = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
+
+            if results is None:
+                return
+
+            row = results.first()
+
+            if row and row['num']:
+                query = "DELETE FROM " + table + " WHERE itemID = '0' or fitID IS NULL OR fitID = '' or fitID = '0'"
+                delete = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
+                logger.error("Database corruption found. Cleaning up %d records.", delete.rowcount)
+
+    @staticmethod
+    def NullDamageTargetPatternValues(saveddata_engine):
+        # Find patterns that have null values
+        # See issue #954
+        for profileType in ['damagePatterns', 'targetResists']:
+            for damageType in ['em', 'thermal', 'kinetic', 'explosive']:
+                logger.debug("Running database cleanup for null %s values. (%s)", profileType, damageType)
+                query = "SELECT COUNT(*) AS num FROM " + profileType + " WHERE " + damageType + "Amount IS NULL OR " + damageType + "Amount = ''"
+                results = DatabaseCleanup.ExecuteSQLQuery(saveddata_engine, query)
+
+                if results is None:
+                    return
+
+                row = results.first()
+
+                if row and row['num']:
+                    query = "UPDATE '" + profileType + "' SET '" + damageType + "Amount' = '0' WHERE " + damageType + "Amount IS NULL OR emAmount = ''"
```

Comment:
looks like `emAmount` was left here, but it's easy enough for me to change after merging.`